Sharpmake/Util.cs

```diff
@@ -98,6 +98,21 @@ public static int GetDeterministicHashCode(this string str)
             }
         }
 
+        private static string GetTextTemplateDirectiveParam(string[] templateText, string directive, string paramName)
+        {
+            Regex regex = new Regex(@"<#@\s*?" + directive + @"\s+?.*?" + paramName + @"=""(?<paramValue>.*?)"".*?#>");
```

Comment:
Since the regex depends on the inputs, the best I could do was cache it in a dictionary on demand. Upgraded the GetOrAdd extension to create on demand as well while I was there.